modules/nw-multinetwork-sriov.adoc

```diff
@@ -40,275 +64,456 @@ and device ID 0x1015
+have been applied until all the Pods in `sriov-network-operator` namespace are
+in `Running` status.
+=====
+
+The following is an example of a typical Custom Resource for
+`SriovNetworkNodePolicy`.
 
-. Create a YAML file for the Custom Resource (CR) with SR-IOV configuration. The
-`name` field in the following CR  has the value `sriov-conf`.
-+
 [source,yaml]
 ----
-apiVersion: "k8s.cni.cncf.io/v1"
-kind: NetworkAttachmentDefinition
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
 metadata:
-  name: sriov-conf
-  annotations:
-    k8s.v1.cni.cncf.io/resourceName: openshift.io/sriov <1>
+  name: policy-example <1>
+  namespace: sriov-network-operator <2>
 spec:
-  config: '{
-      "type": "sriov", <2>
-      "name": "sriov-conf",
-      "ipam": {
-        "type": "host-local",
-        "subnet": "10.56.217.0/24",
-        "routes": [{
-          "dst": "0.0.0.0/0"
-        }],
-        "gateway": "10.56.217.1"
-      }
-    }'
+  resourceName: sriov <3>
+  nodeSelector:
+    feature.node.kubernetes.io/network-sriov.capable: "true" <4>
+  priority: 99 <5>
+  mtu: 9000 <6>
+  numVfs: 16 <7>
+  nicSelector: <8>
+    vendor: "15b3" <8>
+    deviceID: "" <9>
+    pfName: ["eno3", "eno4"] <10>
+    rootDevices: ["0000:02:00.0", "0000:02:00.1"] <11>
+  deviceType: netdevice <12>
+  isRdma: false <13>
 ----
-+
-<1> `k8s.v1.cni.cncf.io/resourceName` annotation is set to `openshift.io/sriov`.
-<2> `type` is set to `sriov`.
+<1> The name of the CR.
+<2> The namespace of the CR, this must be in the same namespace of operator.
+<3> The resource name of SR-IOV device plug-in. Prefix `openshift.io/` will be
+added when it's referred in Pod spec. It's allowed to create multiple CRs of
+`SriovNetworkNodePolicy` for one resource name.
+<4> The node selector to select which node to be configured. User can choose to
+label the nodes manually or with tools like Kubernetes Node Feature Discovery.
+Only SR-IOV network devices on selected nodes will be configured. And the SR-IOV
+CNI plug-in and device plug-in will be only deployed on selected nodes.
+<5> The priority of the policy,  the larger number gets lower priority.
+Range from 0 to 99.
+<6> The MTU of the virtual functions. Range from 1 to 9000. Leave it blank if
+you don't need to change the MTU.
+<7> The number of the virtual functions is to be created for each SR-IOV
+physical network device.
+<8> The NIC selector selects the NIC to be configured. You don't have to specify
+all the fields. However it is recommended to select the NICs as specifically as
+possible, in order to minimize the possibility of overlapping cross policies.
+And if you specify both `pfName` and `rootDevices` at the same time, which is
+recommended, please make sure they point to the identical devices.
+<9> The vendor hex code of SR-IOV device. Allowed value "8086", "15b3".
+<10> The SR-IOV PF device hex code of SR-IOV device. Allowed value "158b", "1015", "1017".
```

Comment:
This should be VF device IDs according to my testing, otherwise device plugin will expose PF as resource.

Reply:
And we may also want to list a mapping of VF IDs to supported NICs so that user knows what to configure in node policy custom resource.

Reply:
> This should be VF device IDs according to my testing, otherwise device plugin will expose PF as resource.

It's a bug, the deviceID shall always be set with the PF's ID. The operator shall convert it to the VF device ID when generating the device plugin configuration, if the numVFs is set.

Reply:
@zhaozhanqi please help to confirm if that's the case in your testing.

Reply:
this should be PF deviceID since we do not know the vf device id yet before the nodenetworkpolicy is created.
```
   - Vfs:
        - deviceID: 154c
          driver: iavf
          mtu: 1500
          pciAddress: '0000:3b:02.0'
          vendor: '8086'
        - deviceID: 154c
          driver: iavf
          mtu: 1500
          pciAddress: '0000:3b:02.1'
          vendor: '8086'
        - deviceID: 154c
          driver: iavf
          mtu: 1500
          pciAddress: '0000:3b:02.2'
          vendor: '8086'
        - deviceID: 154c
          driver: iavf
          mtu: 1500
          pciAddress: '0000:3b:02.3'
          vendor: '8086'
        - deviceID: 154c
          driver: iavf
          mtu: 1500
          pciAddress: '0000:3b:02.4'
          vendor: '8086'
```

Reply:
`oc get configmap -n sriov-network-operator device-plugin-config -o yaml
apiVersion: v1
data:
  config.json: '{"resourceList":[{"resourceName":"intelnetdevice","IsRdma":false,"selectors":{"vendors":["8086"],"drivers":["iavf","mlx5_core","i40evf","ixgbevf"],"pfNames":["ens1f0"]}}]}'
kind: ConfigMap
`